sdks/python/apache_beam/runners/interactive/pipeline_analyzer.py

```diff
@@ -138,32 +143,51 @@ def _analyze_pipeline(self):
                   self._context.to_runner_api().windowing_strategies)
 
     self._pipeline_proto_to_execute = pipeline_to_execute
-    self._top_level_referenced_pcollection_ids = top_level_referenced_pcollection_ids # pylint: disable=line-too-long
+    self._top_level_referenced_pcoll_ids = top_level_referenced_pcoll_ids
     self._top_level_required_transforms = top_level_required_transforms
 
+  # -------------------------------------------------------------------------- #
   # Getters
+  # -------------------------------------------------------------------------- #
+
+  def pipeline_info(self):
+    """Return PipelineInfo of the original pipeline.
+    """
+    return self._pipeline_info
 
   def pipeline_proto_to_execute(self):
     """Returns Pipeline proto to be executed.
     """
     return self._pipeline_proto_to_execute
 
-  def top_level_referenced_pcollection_ids(self):
-    """Returns an array of top level referenced PCollection IDs.
+  def tl_referenced_pcoll_ids(self):
+    """Returns a set of PCollection IDs referenced by top level PTransforms.
     """
-    return self._top_level_referenced_pcollection_ids
+    return self._top_level_referenced_pcoll_ids
 
-  def top_level_required_transforms(self):
-    """Returns a dict mapping ID to proto of top level PTransforms.
+  def tl_required_trans_ids(self):
+    """Returns a set of required top level PTransform IDs.
     """
-    return self._top_level_required_transforms
+    return self._top_level_required_transforms.keys()
```

Comment:
We should return list(self._top_level_required_transforms) to avoid the lint complaint.

Reply:
Also same change maybe needed in line 124 for Python 3 compatibility.

Reply:
I'm trying to do the change now

Reply:
Thanks!